File: ros2_ws/src/activity2_cpp_pkg/src/number_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include <random>
#include "example_interfaces/msg/int64.hpp"

class NumberPublisherNode : public rclcpp::Node
{
public:
    NumberPublisherNode() : Node("number_publisher")
    {
        // Define the range for random integers
        std::uniform_int_distribution<> distrib(1, 100);

        publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number", 10);
        timer_ = this->create_wall_timer(std::chrono::milliseconds(500), std::bind(&NumberPublisherNode::publishNumber, this));

        RCLCPP_INFO(this -> get_logger(), "Number publisher has been started.");
    }

private:
    std::mt19937 gen; // Motor de generación de números aleatorios
    std::uniform_int_distribution<> distrib;
    std::string name_;
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;

    void publishNumber()
    {
        auto msg = example_interfaces::msg::Int64();
        msg.data = distrib(gen);
        publisher_->publish(msg);
    }
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NumberPublisherNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: ros2_ws/src/activity2_py_pkg/activity2_py_pkg/number_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
import random

from example_interfaces.msg import Int64

class NumberPublisherNode(Node):
    def __init__(self):
        super().__init__("number_publisher")
        self.name_ = 'ONE'
        self.publisher_ = self.create_publisher(Int64, 'number', 10)

        self.timer_ = self.create_timer(0.5, self.publishNumber)
        self.get_logger().info(f"Number publisher node {self.name_} running!")


    def publishNumber(self):
        msg = Int64()
        msg.data = random.randint(1, 100)
        self.publisher_.publish(msg)




def main(args=None):
    rclpy.init(args=args)
    node = NumberPublisherNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()
